### PR TITLE
feat: support read sn and read s33

### PR DESCRIPTION
**Pull request by little-willy**
In our current code of file manager, we have two method, `readS32` and `readS64`, which both read a variable length integer from the file. Actually, they seems to share the same read logic as described in [sN](https://webassembly.github.io/spec/core/binary/values.html#binary-int). Besides, as proposed in #2029, we may support reading `s33`, which also share the same read logic. 

Therefore, in this PR, I support reading `sN` as described in [sN](https://webassembly.github.io/spec/core/binary/values.html#binary-int), where N becomes a static constant template variable, and enables static usage check to ensure correctness. `readS32`, `readS64` and `readS33` are simply specialization of `readSN`.

### CODE REVIEW

**Comment by juntao** [comment]
Hi @little-willy 

Thank you for your contribution!

CNCF requires its projects to have DCO signatures for all contributions. Do you mind fixing the DCO on your commits? Thank you!

https://www.secondstate.io/articles/dco/

File: lib/loader/filemgr.cpp [hunk 1]
```diff
@@ -275,8 +287,8 @@ Expect<std::string> FileMgr::readName() {
   if (unlikely(Status != ErrCode::Value::Success)) {
     return Unexpect(Status);
   }
-  // If UTF-8 validation or readU32() or readBytes() failed, the last succeeded
-  // reading offset will be at the start of `Name`.
+  // If UTF-8 validation or readU32() or readBytes() failed, the last
+  // succeeded reading offset will be at the start of `Name`.
```
Comment (q82419) [inline lib/loader/filemgr.cpp:291]: This line is not necessary to be changed.

File: lib/loader/filemgr.cpp [hunk 2]
```diff
@@ -411,7 +423,8 @@ Expect<void> FileMgr::jumpContent() {
   return {};
 }
 
-// Helper function for reading number of bytes. See "include/loader/filemgr.h".
+// Helper function for reading number of bytes. See
+// "include/loader/filemgr.h".
```
Comment (q82419) [inline lib/loader/filemgr.cpp:427]: This line is not necessary to be changed.

**Review by q82419 — CHANGES_REQUESTED** [review]
Can you help to fix the loading of block type [here](https://github.com/WasmEdge/WasmEdge/blob/master/lib/loader/ast/instruction.cpp#L188)?

Thanks!

File: lib/loader/filemgr.cpp [hunk 3]
```diff
@@ -144,84 +144,96 @@ Expect<uint64_t> FileMgr::readU64() {
...
-  uint64_t Offset = 0;
-  Byte Byte = 0x80;
-  while (Byte & 0x80) {
-    if (unlikely(Offset >= 64)) {
-      Status = ErrCode::Value::IntegerTooLong;
-      return Unexpect(Status);
-    }
-    if (auto Res = testRead(1); unlikely(!Res)) {
+    WasmEdge::Byte Byte;
+    if (auto Res = readByte(); unlikely(!Res)) {
```
Comment (q82419) [inline lib/loader/filemgr.cpp:173]: Please use the `testRead(1)` and `Byte = Data[Pos++]` mechanism as the origin code.
This is for printing the correct error message of the loading offset when error occurs.

**Review by q82419 — CHANGES_REQUESTED** [review]

**Comment by little-willy** [comment]
@q82419 Thanks for reviewing. I have fixed the comment and added unit tests for `readS33`. PTAL.

> Can you help to fix the loading of block type [here](https://github.com/WasmEdge/WasmEdge/blob/master/lib/loader/ast/instruction.cpp#L188)?

Can I include this fix in a later PR that will fix #2029 totally and merge this PR first?

**Review by q82419 — APPROVED** [review]

**Comment by q82419** [comment]
> @q82419 Thanks for reviewing. I have fixed the comment and added unit tests for `readS33`. PTAL.
> 
> > Can you help to fix the loading of block type [here](https://github.com/WasmEdge/WasmEdge/blob/master/lib/loader/ast/instruction.cpp#L188)?
> 
> Can I include this fix in a later PR that will fix #2029 totally and merge this PR first?

Sure.
Thanks for your contribution!

**Pull request merged**